test/lib/dockerfile/index.spec.ts

```diff
@@ -52,6 +52,45 @@ describe("base image parsing", () => {
     });
   });
 
+  it.each`
+    dockerfile
+    ${"ARG A" + EOL + "FROM image" + EOL + "FROM ${A}"}
+    ${"ARG A" + EOL + "FROM image" + EOL + "FROM ${A}:1.0.0"}
+    ${"ARG A" + EOL + "FROM image" + EOL + "FROM ${A}@sha256:abcd1234"}
+    ${"ARG A" + EOL + "FROM image" + EOL + "FROM image:${A}"}
+    ${"ARG A" + EOL + "FROM image" + EOL + "FROM image@${A}"}
+  `(
+    "does not detect unresolvable final stage: $dockerfile",
+    ({ dockerfile }) => {
+      const result = getDockerfileBaseImageName(
+        DockerfileParser.parse(dockerfile),
+      );
+
+      expect(result.baseImage).toBeUndefined();
+      expect(result).toEqual({
+        error: {
+          code: DockerFileAnalysisErrorCode.BASE_IMAGE_NON_RESOLVABLE,
+        },
+      });
+    },
+  );
+
+  it.each`
```

Comment:
do we also have a test case for an unresolvable stage, where the final image depends on a resolvable image ( hope that makes sense)?

```dockerfile
ARG A

FROM image:tag as base

FROM ${A} # not resolved

FROM base # resolved 
```

Reply:
Makes perfect sense. Well spotted 😄